File: app/src/main/java/com/example/myapplication2/Kotlin/lesson2/Singleton.java
package com.example.myapplication2.Kotlin.lesson2;

/**
 * Паттерн не нужный в андроиде
 */
public class Singleton {
    private static Singleton sSingleton;

    public static Singleton getInstance(){
        if (sSingleton == null){
            sSingleton = new Singleton();
        }
        return sSingleton;
    }

    private Singleton(){
        System.out.println("constructor singleton called");
    }

}

File: app/src/main/java/com/example/myapplication2/Kotlin/lesson2Kt/Singleton.kt
package com.example.myapplication2.Kotlin.lesson2Kt

//синглтон, это похоже на статику
internal object Main {
    @JvmStatic
    fun main(array: Array<String>){
        Singleton.increment()
        Singleton.increment()
        Singleton.print()
    }
}

internal object Singleton{
    private var x = 0

    fun increment(){
        x++
    }

    fun print(){
        print(x)
    }
}